### PR TITLE
feat(cce/cluster): support multi eni subnet IDs, and updating the eni subnets

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cce' TESTARGS='-run TestAccCluster_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cce -v -run TestAccCluster_basic -timeout 360m -parallel 4
=== RUN   TestAccCluster_basic
=== PAUSE TestAccCluster_basic
=== CONT  TestAccCluster_basic
--- PASS: TestAccCluster_basic (530.33s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cce       530.412s

make testacc TEST='./huaweicloud/services/acceptance/cce' TESTARGS='-run TestAccCluster_turbo'==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cce -v -run TestAccCluster_turbo -timeout 360m -parallel 4
=== RUN   TestAccCluster_turbo
=== PAUSE TestAccCluster_turbo
=== CONT  TestAccCluster_turbo
--- PASS: TestAccCluster_turbo (586.29s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cce       586.370s
```

### CODE REVIEW

File: docs/resources/cce_cluster.md [hunk 1]
```diff
@@ -271,6 +268,8 @@ In addition to all arguments above, the following attributes are exported:
 
 * `security_group_id` - Security group ID of the cluster.
 
+* `eni_subnet_cidr` - The ENI network segment.
```
Comment (Lance52259) [inline docs/resources/cce_cluster.md]: missing example or description about multiple cidrs

File: huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_cluster_test.go [hunk 2]
```diff
@@ -451,24 +463,100 @@ func testAccCluster_turbo(rName string) string {
...
   vpc_id                 = huaweicloud_vpc.test.id
   subnet_id              = huaweicloud_vpc_subnet.test.id
   container_network_type = "eni"
-  eni_subnet_id          = huaweicloud_vpc_subnet.eni_test.ipv4_subnet_id
-  eni_subnet_cidr        = huaweicloud_vpc_subnet.eni_test.cidr
+  eni_subnet_id          = join(",",[
+    huaweicloud_vpc_subnet.eni_test_1.ipv4_subnet_id,
+    huaweicloud_vpc_subnet.eni_test_2.ipv4_subnet_id
+  ])
+}
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_cluster_test.go]: how about
```
resource "huaweicloud_vpc_subnet" "eni_test" {
  count = 2

  vpc_id = huaweicloud_vpc.test.id
  name  = "%[2]s-eni-${count.index}"
  cidr      = cidrsubnet(huaweicloud_vpc.test.cidr, 4, count.index)
  gateway_ip = cidrhost(cidrsubnet(huaweicloud_vpc.test.cidr, 4, count.index), 1)
}
```
and refer in ECS instnace:
```
eni_subnet_id = join(",", huaweicloud_vpc_subnet.eni_test[*].id)
```

File: huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_cluster_test.go [hunk 3]
```diff
@@ -451,24 +463,100 @@ func testAccCluster_turbo(rName string) string {
...
+
+locals {
+  eni_subnet_ids = split(",",huaweicloud_cce_cluster.test.eni_subnet_id)
+}
+
+output "eni_subnet_id_1" {
+  value = (local.eni_subnet_ids[0] == huaweicloud_vpc_subnet.eni_test_1.ipv4_subnet_id)
 }
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_cluster_test.go]: one output variable also can check the difference for all subnet IDs
```
output "is_eni_subnet_id_different" {
  value = length(setsubtract(split(",",huaweicloud_cce_cluster.test.eni_subnet_id), huaweicloud_vpc_subnet.eni_test[*].id)) != 0
}
```

**Comment by Jason-Zhang9309** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/cce' TESTARGS='-run TestAccCluster_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cce -v -run TestAccCluster_basic -timeout 360m -parallel 4
=== RUN   TestAccCluster_basic
=== PAUSE TestAccCluster_basic
=== CONT  TestAccCluster_basic
--- PASS: TestAccCluster_basic (521.84s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cce       521.943s

make testacc TEST='./huaweicloud/services/acceptance/cce' TESTARGS='-run TestAccCluster_turbo'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cce -v -run TestAccCluster_turbo -timeout 360m -parallel 4
=== RUN   TestAccCluster_turbo
=== PAUSE TestAccCluster_turbo
=== CONT  TestAccCluster_turbo
--- PASS: TestAccCluster_turbo (606.19s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cce       606.270s
```

File: huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_cluster_test.go (hunk 3 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_cluster_test.go]: sure

File: huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_cluster_test.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_cluster_test.go]: sure

File: docs/resources/cce_cluster.md (hunk 1 above)
Comment (Jason-Zhang9309) [reply docs/resources/cce_cluster.md]: added

File: huaweicloud/services/cce/resource_huaweicloud_cce_cluster.go [hunk 4]
```diff
@@ -626,7 +636,7 @@ func resourceClusterRead(_ context.Context, d *schema.ResourceData, meta interfa
 		d.Set("highway_subnet_id", n.Spec.HostNetwork.HighwaySubnet),
 		d.Set("container_network_type", n.Spec.ContainerNetwork.Mode),
 		d.Set("container_network_cidr", flattenContainerNetworkCidrs(n.Spec.ContainerNetwork)),
-		d.Set("eni_subnet_id", n.Spec.EniNetwork.SubnetId),
+		d.Set("eni_subnet_id", flattenEniSubnetID(n.Spec.EniNetwork)),
 		d.Set("eni_subnet_cidr", n.Spec.EniNetwork.Cidr),
```
Comment (ShiChangkuo) [inline huaweicloud/services/cce/resource_huaweicloud_cce_cluster.go:640]: what's the value of eni_subnet_id when we specify multi eni_subnet_id ?

Comment (Jason-Zhang9309) [reply huaweicloud/services/cce/resource_huaweicloud_cce_cluster.go:640]: the description of eni_subnet_cidr is updated

**Pull request merged**